core/ResManager/plKeyCollector.cpp

```diff
@@ -242,6 +234,7 @@ void plKeyCollector::ChangeLocation(const plLocation& from, const plLocation& to
             for (size_t j=0; j<keys[from][i].size(); j++) {
                 keys[to][i][begin+j] = keys[from][i][j];
                 keys[to][i][begin+j]->setLocation(to);
+                keys[to][i][begin+j]->setID(begin+j+1);
```

Comment:
If the keys are not necessarily always contiguous, doesn't this need to be based on the ID of the last key in the old list rather than the position in the array?

Reply:
The intention is to force the list to be contiguous on read in so libHSPlasma's two assumptions are always valid for any pages it has completely read in. Of course, that means you're right that this code is wrong if the page has not been read in. But what does it mean to move a key to an not loaded page, and do we care? 😅

Reply:
Good point, since we're fixing the key IDs on load (and on new key creation), they should always be contiguous by this point in the code...